### PR TITLE
feat(jsii): add excludeTypescript config option

### PR DESCRIPTION
**Pull request by polothy**
Hello 👋 

Working on a CDK constructs project where I have a setup like:

- `src/construct-one/`
- `src/construct-two/`
- etc...

Thought it would be a neat idea to put the tests inside each construct, so in `src/construct-one/test` for example. But when JSII compiled the project for JS/TypeScript, it included the `test` directory. I was able to get JSII to ignore the `test` directory by adding this to my `package.json`:

```json
{
  "jsii": {
    "excludeTypescript": ["src/**/test/*.ts"]
  }
}
```

So, this PR is to add this configuration option to `projen`.

Link to JSII docs: https://github.com/aws/jsii/blob/main/gh-pages/content/user-guides/lib-author/configuration/index.md#the-jsii-section

I'm very open to suggestions (even if it is that my location of the `test` directory is a bad idea) and let me know if the PR can be improved in any way.  In addition to the unit test lines, I did test by creating a new project, updating the `.projenrc.js` and regenerating the project.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by polothy** [comment]
If there are any PR build failures, I'll check them out tomorrow, cheers!

**Comment by polothy** [comment]
Hrm, maybe this isn't the best idea.  Now I'm having a problem where `projen` is re-creating the root `test` directory with sample test code.  The option to control this is in the `TypeScriptProjectOptions`:

https://github.com/projen/projen/blob/a8bef4c91fdb8e6c9268d74a90000da21492ce72/src/typescript.ts#L97-L101

Should I copy this option to the `JsiiProjectOptions` so it can be passed to the `TypeScriptProject` constructor? Or abandon this whole idea?

**Comment by eladb** [comment]
> Should I copy this option to the JsiiProjectOptions so it can be passed to the TypeScriptProject constructor? Or abandon this whole idea?

Copy.

File: src/jsii-project.ts [hunk 1]
```diff
@@ -103,6 +103,15 @@ export interface JsiiProjectOptions extends NodeProjectOptions {
    * @default ".compatignore"
    */
   readonly compatIgnore?: string;
+
+  /**
+   * Accepts a list of glob patterns. Files matching any of those patterns will be excluded from the TypeScript
```
[the remark this answers is not in the document]
Comment (polothy) [reply src/jsii-project.ts:108]: Yes, copy/paste fail from JSII docs hah

File: src/jsii-project.ts [hunk 2]
```diff
@@ -197,16 +206,18 @@ export class JsiiProject extends TypeScriptProject {
...
 
-    this.addFields({
-      jsii: {
-        outdir: 'dist',
-        targets,
-        tsc: {
-          outDir: libdir,
-          rootDir: srcdir,
-        },
+    let jsii: any = {
```
[the remark this answers is not in the document]
Comment (polothy) [reply src/jsii-project.ts]: Thanks, thought I had to use `let` because I modified the value.  More to learn!

**Comment by polothy** [comment]
Pushed a new commit to address discussion and review 👍 🚀

**Review by eladb — APPROVED** [review]

**Comment by polothy** [comment]
Thanks!

**Comment by eladb** [comment]
@polothy thank you!

**Pull request merged**